inductiva/inductiva
# Add threads per core

[#297](https://github.com/inductiva/tasks/issues/297)
Client implementation for https://github.com/inductiva/inductiva-web-api/pull/1190

This PR adds an option to define the number of threads_per_core, default is 2 can be changed to 1. This is the same behavior of the google cloud.

For more info, check:
https://cloud.google.com/compute/docs/instances/set-threads-per-core



How to test: 
```python
import inductiva

mg = inductiva.resources.MachineGroup(machine_type="c2-standard-4",
                                      num_machines=1,
                                      threads_per_core=1)

mg.start()
```
And then check on google cloud for the vCPUs to core ratio of the VM.

Note:  mg.n_vcpus does not correctly display the number of vCPUS. The implementation currently only checks the vCPUs of the machine_type. It does not actually query the machine.

Note2: "In the Google Cloud console, when the number of threads per core is set to 1, the CPU utilization metric can only scale up to a maximum of 50%. When the number of threads per core is set to 2, the CPU utilization metric can scale up to 100%."
